### ISSUE TITLE
[3.6.4] AttributeError: module 'regex' has no attribute 'Pattern'

### ISSUE THREAD

**Issue by astariul** [nltk/nltk#2833]
Since I upgraded `nltk` from `3.6.2` to `3.6.4`, the following code :

```python
from nltk.tokenize.casual import TweetTokenizer, casual_tokenize
```

raises the following error :

> Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/user/miniconda3/envs/test/lib/python3.7/site-packages/nltk/__init__.py", line 137, in <module>
    from nltk.text import *
  File "/home/user/miniconda3/envs/test/lib/python3.7/site-packages/nltk/text.py", line 29, in <module>
    from nltk.tokenize import sent_tokenize
  File "/home/user/miniconda3/envs/test/lib/python3.7/site-packages/nltk/tokenize/__init__.py", line 65, in <module>
    from nltk.tokenize.casual import TweetTokenizer, casual_tokenize
  File "/home/user/miniconda3/envs/test/lib/python3.7/site-packages/nltk/tokenize/casual.py", line 272, in <module>
    class TweetTokenizer:
  File "/home/user/miniconda3/envs/test/lib/python3.7/site-packages/nltk/tokenize/casual.py", line 357, in TweetTokenizer
    def WORD_RE(self) -> regex.Pattern:
**AttributeError: module 'regex' has no attribute 'Pattern'**

**Comment by purificant** [comment]
Had a quick look and was unable to reproduce.

```
pip install --upgrade pip && pip install nltk==3.6.4
```

```python
>>> import nltk
>>> nltk.__version__
'3.6.4'
>>> from nltk.tokenize.casual import TweetTokenizer, casual_tokenize
>>> 
```

Edit: on python 3.9.7 / 3.7.12

**Comment by tomaarsen** [comment]
~~It seems that the `re.Pattern` (and `regex.Pattern`) attributes were only introduced starting from Python 3.8+. How our tests didn't catch this - I have no idea. I'll make a fix.~~

Edit: This doesn't seem to be the issue - I think `regex.Pattern` is not defined on older versions of `regex`. A fix is updating `regex` to the newest version. See https://bitbucket.org/mrabarnett/mrab-regex/commits/26a320f29b8266562a67dfa2243feda1a8db40ac for more information.

**Comment by tomaarsen** [comment]
~~@astariul For the time being, please use the develop branch instead of 3.6.4. This command ought to do it:~~

See my last comment - you need to update `regex`. I've made a PR to ensure that `regex` will always be up to date from now on.
```
pip install -U regex
```

**Comment by astariul** [comment]
Awesome thanks @tomaarsen (and @purificant) for the lightning fast answers !

**Comment by purificant** [comment]
I am able to confirm that once I downgrade `regex` package, I can also reproduce the issue. Could not reproduce with a fresh nltk install in a fresh venv before in different python versions.

Thank you for reporting the issue @astariul and thank you @tomaarsen for tracking down the root cause / fix.

**Comment by tomaarsen** [comment]
Version 3.6.4 has been yanked from PyPI as a result of this issue. That said, you can still use it if desired - as long as you use a high enough `regex` version. See comments above for more information.

**Comment by tomaarsen** [comment]
This has been fixed in the new release 3.6.5. Thank you for raising this issue.
